core/src/main/java/io/parsingdata/metal/expression/value/Value.java

```diff
@@ -16,69 +16,20 @@
 
 package io.parsingdata.metal.expression.value;
 
-import static io.parsingdata.metal.Util.bytesToHexString;
-import static io.parsingdata.metal.Util.checkNotNull;
-import static io.parsingdata.metal.encoding.Encoding.DEFAULT_ENCODING;
-
 import java.math.BigInteger;
 import java.util.BitSet;
-import java.util.Objects;
 
-import io.parsingdata.metal.Util;
 import io.parsingdata.metal.data.Slice;
-import io.parsingdata.metal.encoding.ByteOrder;
 import io.parsingdata.metal.encoding.Encoding;
-import io.parsingdata.metal.encoding.Sign;
-
-public class Value {
-
-    public static final Value NOT_A_VALUE = new Value(Slice.createFromBytes(new byte[]{}), DEFAULT_ENCODING);
-    public static final BigInteger TO_STRING_BYTE_COUNT = BigInteger.valueOf(4);
-
-    public final Slice slice;
-    public final Encoding encoding;
-
-    public Value(final Slice slice, final Encoding encoding) {
-        this.slice = checkNotNull(slice, "slice");
-        this.encoding = checkNotNull(encoding, "encoding");
-    }
-
-    public byte[] getValue() {
-        return slice.getData();
-    }
-
-    public BigInteger getLength() {
-        return slice.length;
-    }
-
-    public BigInteger asNumeric() {
-        return encoding.sign == Sign.SIGNED ? new BigInteger(encoding.byteOrder.apply(getValue()))
-                                            : new BigInteger(1, encoding.byteOrder.apply(getValue()));
-    }
-
-    public String asString() {
-        return new String(getValue(), encoding.charset);
-    }
-
-    public BitSet asBitSet() {
-        return BitSet.valueOf(encoding.byteOrder == ByteOrder.BIG_ENDIAN ? ByteOrder.LITTLE_ENDIAN.apply(getValue()) : getValue());
-    }
-
-    @Override
-    public String toString() {
-        return "0x" + bytesToHexString(slice.getData(TO_STRING_BYTE_COUNT)) + (getLength().compareTo(TO_STRING_BYTE_COUNT) > 0 ? "..." : "");
-    }
 
-    @Override
-    public boolean equals(final Object obj) {
-        return Util.notNullAndSameClass(this, obj)
-            && Objects.equals(slice, ((Value)obj).slice)
-            && Objects.equals(encoding, ((Value)obj).encoding);
-    }
+public interface Value {
 
-    @Override
-    public int hashCode() {
-        return Objects.hash(getClass(), slice, encoding);
-    }
+    Slice getSlice();
```

Comment:
We discussed this before, I think, but I'd like to see shorter names here, and just use `slice()`, `encoding()`, `value()`, and `length()`. We can do this in a separate issue.

Reply:
I agree. Also add spaces between the methods? :smile:

Reply:
I fixed both the names and the spacing. In some other places as well by the way, so it should be universal now that there is a newline _between_ methods :)